arch/arm/src/armv6-m/arm_svcall.c

```diff
@@ -92,7 +92,7 @@ static void dispatch_syscall(void)
     " ldr r4, =g_stublookup\n"                      /* R4=The base of the stub lookup table */
     " lsl r0, r0, #2\n"                             /* R0=Offset of the stub for this syscall */
     " ldr r4, [r4, r0]\n"                           /* R4=Address of the stub for this syscall */
-    " blx r5\n"                                     /* Call the stub (modifies lr) */
+    " blx r4\n"                                     /* Call the stub (modifies lr) */
```

Comment:
how about other m arch?

Reply:
> how about other m arch?

other arch do not have this issue, the following is the implementation in armv7-m and armv8-m:
```
      " ldr ip, =g_stublookup\n"                      /* R12=The base of the stub lookup table */
      " ldr ip, [ip, r0, lsl #2]\n"                   /* R12=The address of the stub for this syscall */
      " blx ip\n"                                     /* Call the stub (modifies lr) */
```
which correct handling the sub-routine calling procedure

Reply:
Ok